### ISSUE TITLE
Confuses functions with the same name, but different namespace (Origin: bugzilla #769411)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#6040]
status NEW severity *major* in component *general* for *---*
Reported in version *1.8.10* on platform *Other*
Assigned to: Dimitri van Heesch

Original attachment names and IDs:
- _[doxybug.tgz](https://sourceforge.net/projects/doxygen/files/attachments/bug769411_332510_doxybug.tgz/download/download)_ (ID 332510)

On 2016-08-02 09:14:08 +0000, Akim Demaille wrote:

    Created attachment 332510
    Source and Doxyfile
    
    The following source:
    
        #pragma once
        
        namespace vcsn
        {
          namespace dyn
          {
            /// A reversed trie-like automaton (multiple initial states,
            /// single final state) automaton to accept \a p.
            ///
            /// \param p    the polynomial that describes the (finite) series
            automaton cotrie(const polynomial& p);
        
            namespace detail
            {
              /// Bridge.
              template <typename PolynomialSet>
              automaton
              cotrie(const polynomial& poly)
              {}
            }
          }
        } // vcsn::
    
    let's Doxygen confuses both functions 'cotrie':
    
        doxygen: /private/tmp/foo.hh:7: warning: argument 'p' of command @param
    is not found in the argument list of vcsn::dyn::cotrie(const polynomial
    &poly)
        doxygen: /private/tmp/foo.hh:11: warning: The following parameters of
    vcsn::dyn::cotrie(const polynomial &poly) are not documented:
          parameter 'poly'
    
    and the generated documentation does feature this problem.

On 2016-10-10 18:16:41 +0000, Akim Demaille wrote:

    Ping?

On 2016-10-11 13:11:23 +0000, Younes Khoudli wrote:

    The bug is still triggered both with Doxygen 1.8.12 and the latest git
    commit (985faf287), on Archlinux.

**Comment by marcalff** [comment]
The bug still exists in Doxygen 1.8.17:

malff@linux-8edv:bug_namespace> pwd
/home/malff/EXTERNAL/doxygen/bug_namespace

malff@linux-8edv:bug_namespace> doxygen -x
# Difference with default Doxyfile 1.8.17 (b5fa3cd1c6e6240e20d3b80a70e3f04040b32021*)
INPUT                  = foo.cc

malff@linux-8edv:bug_namespace> cat foo.cc

/**
  @file foo.cc
  Some test.
*/

namespace foo {
  /** AAA.
    @param x Some X
    @param y Some Y
    @return something
    @retval 0 passed
    @retval 1 failed
  */
  int bar(int x, int y);
}

/** BBB.
  @param x Some other X
  @param y Some other Y
  @return something else
  @retval 0 passed again
  @retval 1 failed again
*/
int bar(int x, int y);

int foo::bar(int x, int y) {
  return x+y;
}

int bar(int x, int y) {
  return x-y;
}

/** The main. */
int main(int, char*[]) {
  foo::bar(1, 2);
  bar(3, 4);
  return 0;
}

malff@linux-8edv:bug_namespace> doxygen
...
Generating docs for file foo.cc...
/home/malff/EXTERNAL/doxygen/bug_namespace/foo.cc:15: warning: return value '0' of foo::bar has multiple documentation sections
/home/malff/EXTERNAL/doxygen/bug_namespace/foo.cc:15: warning: return value '1' of foo::bar has multiple documentation sections
/home/malff/EXTERNAL/doxygen/bug_namespace/foo.cc:15: warning: argument 'x' from the argument list of foo::bar has multiple @param documentation sections
/home/malff/EXTERNAL/doxygen/bug_namespace/foo.cc:15: warning: argument 'y' from the argument list of foo::bar has multiple @param documentation sections

This also affects MySQL, seen with:
- function compute_sha1_hash()
- function my_sha1::compute_sha1_hash()

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7504.

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by marcalff** [comment]
With a recent build from master, which includes the fix from Albert, I confirm the bug is resolved.

tested with:
```
doxygen --version
1.8.18 (8f71c69d1517f5259652a739830c732983d118aa*)

```

Thank you.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.18.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
